src/App.tsx

```diff
@@ -15,37 +15,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const showList = (arr: string[]) =>
+  arr.map(good => (
+    <li key={good} data-cy="Good">
+      {good}
+    </li>
+  ));
+
+const defaultList = showList(goodsFromServer);
+
+export const App = () => {
+  const [actualList, setActualList] = useState(defaultList);
+  const [actualListArr, setActualListArr] = useState([...goodsFromServer]);
+  const [sortMethod, setSortMethod] = useState<string | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortArray = (arr: string[], method: string): string[] => {
+    let sortedArr;
+
+    switch (method) {
+      case 'alphabetically':
+        sortedArr = [...arr].sort((a, b) => a.localeCompare(b));
+        break;
+      case 'byLength':
+        sortedArr = [...arr].sort((a, b) => a.length - b.length);
+        break;
+      default:
+        sortedArr = arr;
+    }
+
+    return sortedArr;
+  };
+
+  const handleSort = (method: string) => {
+    let sortedArr = sortArray(goodsFromServer, method);
+
+    if (isReversed) {
+      sortedArr = [...sortedArr].reverse();
+    }
+
+    setSortMethod(method);
+    setActualListArr(sortedArr);
+    setActualList(showList(sortedArr));
+  };
+
+  const handleReverse = () => {
+    const reversedArr = [...actualListArr].reverse();
+
+    setIsReversed(!isReversed);
+    setActualListArr(reversedArr);
+    setActualList(showList(reversedArr));
+  };
+
+  const handleReset = () => {
+    setSortMethod(null);
+    setIsReversed(false);
+    setActualListArr([...goodsFromServer]);
+    setActualList(defaultList);
+  };
+
+  const getButtonClass = (method: string) => {
+    if (sortMethod === method && method === 'alphabetically') {
+      return 'button is-info';
+    }
+
+    if (method === 'alphabetically') {
+      return 'button is-info is-light';
+    }
+
+    if (sortMethod === method && method === 'byLength') {
+      return 'button is-success';
+    }
+
+    if (method === 'byLength') {
+      return 'button is-success is-light';
+    }
+
+    return 'button is-light';
+  };
+
+  const reverseButtonClass = `button is-warning ${isReversed ? '' : 'is-light'}`;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => handleSort('alphabetically')}
+          type="button"
+          className={getButtonClass('alphabetically')}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => handleSort('byLength')}
+          type="button"
+          className={getButtonClass('byLength')}
```

Comment:
[UX] - The 'Sort by length' button should also visually indicate when it is active. The use of 'is-success' class for both states may not provide sufficient feedback to the user. Consider modifying the styling to make the active state more distinct.